versioning/BRM/extentmap.cpp

```diff
@@ -269,19 +270,306 @@ FreeListImpl::FreeListImpl(unsigned key, off_t size, bool readOnly) :
 {
 }
 
+/*static*/
+std::mutex ExtentMapIndexImpl::fInstanceMutex_;
+
+/*static*/
+ExtentMapIndexImpl* ExtentMapIndexImpl::fInstance_ = nullptr;
+
+/*static*/
+ExtentMapIndexImpl* ExtentMapIndexImpl::makeExtentMapIndexImpl(unsigned key, off_t size, bool readOnly)
```

Comment:
Can you please clarify, why do we need a keys, since we identify that segment was grown by other process in the current process using size.

Reply:
We don't need them anymore.

Reply:
Just something come to my mind, could we have a situation when we call `grow` for the segment with size 0, so we increase it on 0 bytes, with following sequence of syscalls `munmap`, `fruncate`, `mmap` and finally got the situation when one segment is `remaped` without the size increased and we cannot identify it, because the size is the same.

Reply:
Only if there will be no grabEMIndex or grow called in the code. This can happen only if the programmer violates a "proto" to operate ManagedShmMem